modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java

```diff
@@ -109,6 +110,19 @@ public void transportClientClusterHealth() {
         }
     }
 
+    @Test
+    public void waitingForShards() throws IOException{
+        try (ElasticsearchContainer container = new ElasticsearchContainer(new ImageFromDockerfile()
+            .withFileFromClasspath("data.zip", "elasticsearch-preload/data.zip")
+            .withFileFromClasspath("Dockerfile", "elasticsearch-preload/Dockerfile")
+        ).waitingForShards()) {
+            container.start();
+            Response response = getClient(container).performRequest(new Request("GET", "/_cat/shards"));
```

Comment:
I wouldn’t rewrite the code in the test. I would test desired behavior: querying should work for example. If shards are not ready the query will fail or return partial result - so run an aggregation query.

Reply:
I agree, I should be able to assert I can query or write.

Reply:
I have changed the assert to see if the preload document can be indeed search.